src/main/java/eu/dissco/backend/controller/RestResponseEntityExceptionHandler.java

```diff
@@ -18,44 +19,79 @@ public class RestResponseEntityExceptionHandler extends ResponseEntityExceptionH
 
   @ResponseStatus(HttpStatus.UNPROCESSABLE_ENTITY)
   @ExceptionHandler(JsonProcessingException.class)
-  public ResponseEntity<String> handleJsonException(JsonProcessingException ex) {
-    return ResponseEntity.status(HttpStatus.UNPROCESSABLE_ENTITY).body(ex.getMessage());
+  public ResponseEntity<ExceptionResponseWrapper> handleJsonException(JsonProcessingException e) {
+    var exceptionResponse = new ExceptionResponseWrapper(
+            HttpStatus.UNPROCESSABLE_ENTITY,
+            "Json Processing Exception",
+            e.getMessage()
+    );
+    return ResponseEntity.status(HttpStatus.UNPROCESSABLE_ENTITY).body(exceptionResponse);
   }
 
   @ResponseStatus(HttpStatus.CONFLICT)
   @ExceptionHandler(ConflictException.class)
-  public ResponseEntity<Void> handleException(ConflictException e) {
-    return ResponseEntity.status(HttpStatus.CONFLICT).build();
+  public ResponseEntity<ExceptionResponseWrapper> handleException(ConflictException e) {
+    var exceptionResponse =  new ExceptionResponseWrapper(
+            HttpStatus.CONFLICT,
+            "Conflict",
```

Comment:
Maybe a bit more context. Think this is a conflict exception that only occurs when the id in the object doesn't match the id in the url. So maybe a "Conflicting id's in message" might cover it better

Reply:
And the message here might be null I think

Reply:
Added more detailed message. Right now, the conflicting ids are captured in the logs but not in an exception. Is that something we want to provide in an error object?

Reply:
This exception is parent to 3 exceptions: InvalidIdException, InvalidTypeException, UserExistsException. However, we handle the 3 exceptions at the parent level, so the granularity is lost, and we don't capture any information in a message. Do we need 3 inheritors of this exception?

Reply:
Ok yes, we could do it this way. Or remove the non-arg constructor from the exceptions, forcing it everywhere to have a message for the exception. The granulatiry could then be in the message that is included in the exception.